### PR TITLE
Added more tests to test_vertical_portscans.py

### PR DESCRIPTION
**Pull request by Sekhar-Kumar-Dash**
## Changes proposed

With this pull request, I have added more tests to test_vertical_portscans.py, which will improve the code coverage

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Comment by AlyaGomaa** [comment]
Hello @Sekhar-Kumar-Dash 
can you give me permission to push to your branch? i need to update it with the latest develop in order for the CI to run correctly

File: tests/test_vertical_portscans.py [hunk 1]
```diff
@@ -4,6 +4,24 @@
...
+        TimeWindow,
+        Attacker,
+        ThreatLevel,
+        EvidenceType,
+        IoCType,
+        Direction,
+        IDEACategory,
+        Victim,
+        Proto,
+        Tag
```
Comment (AlyaGomaa) [inline tests/test_vertical_portscans.py]: You're only using Proto from this import list, why import the following? usually your IDE should warn you about this and can even auto fix them for you so you don't have to worry about it

File: tests/test_vertical_portscans.py [hunk 2]
```diff
@@ -208,3 +227,307 @@ def test_check_if_enough_dports_to_trigger_an_evidence(
...
+def test_get_not_established_dst_ips_invalid_protocol(mock_db):
+    vertical_ps = ModuleFactory().create_vertical_portscan_obj(mock_db)
+    profileid = "profile_1.1.1.1"
+    twid = "timewindow0"
+    protocol = "INVALID"
+    state = "Not Established"
+    role = "Client"
+    type_data = "IPs"
+    direction = "Dst"
+
```
Comment (AlyaGomaa) [inline tests/test_vertical_portscans.py:449]: The above 3 lines are unused local variables. usually your IDe should warn you about them or mark them in gray.

File: tests/test_vertical_portscans.py [hunk 3]
```diff
@@ -208,3 +227,307 @@ def test_check_if_enough_dports_to_trigger_an_evidence(
...
+    profileid = "profile_1.1.1.1"
+    twid = "timewindow0"
+
+    mock_db.get_data_from_profile_tw.side_effect = Exception("Database error")
+
+    with pytest.raises(Exception):
+        vertical_ps.get_not_established_dst_ips(protocol, state, profileid, twid)
+        
+def test_check_overlapping_dstports(mock_db):
+    vertical_ps = ModuleFactory().create_vertical_portscan_obj(mock_db)
```
Comment (AlyaGomaa) [inline tests/test_vertical_portscans.py:498]: Hey @Sekhar-Kumar-Dash I'm having trouble understanding this test
why are you testing overlapping ports? what can go wrong here?
Also i recommend using function docstrings so it's easier to know the answer to the above questions without having to ask you every time :)

**Review by AlyaGomaa — COMMENTED** [review]
Hello @Sekhar-Kumar-Dash Thank you so much for your effort!
I see you spent so much time understanding the code to come up with that many test cases, we appreciate it.

All my comments are related to code style and formatting, i added 1 commit with refactoring that you could learn from, small things like line length and using mark.parametrize could make it easier for anyone reading your code.

Thanks again!:)

File: tests/test_vertical_portscans.py (hunk 1 above)
Comment (Sekhar-Kumar-Dash) [reply tests/test_vertical_portscans.py]: Sorry about that. I am actually using VS Code now, so this feature must be disabled. I will fix this mistake and update the pull request.

File: tests/test_vertical_portscans.py (hunk 2 above)
Comment (Sekhar-Kumar-Dash) [reply tests/test_vertical_portscans.py:449]: Yes, it is actually getting displayed in a light color. Sorry, I missed that earlier. I am fixing it.

File: tests/test_vertical_portscans.py (hunk 3 above)
Comment (Sekhar-Kumar-Dash) [reply tests/test_vertical_portscans.py:498]: hey @AlyaGomaa The i have included this test is to ensure that the VerticalPortscan module correctly aggregates the packet counts for overlapping destination ports when scanning both TCP and UDP protocols.  it is not a critical test we can remove it if you dont want it 

am i getting it wrong??
Also thanks for the docstrings suggestion   i will keep that in mind for future unittests PRs

**Pull request merged**